src/reactor.cc

```diff
@@ -0,0 +1,333 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+Reactor::Reactor(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  fuel_tracker.Init({&tritium_storage}, fuel_limit);
+  blanket_tracker.Init({&blanket}, blanket_limit);
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  sequestered_tritium->Decay(context()->time());
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    if (surplus > 0.0) {
+
+      tritium_excess.Push(tritium_storage.Pop(surplus));
+      CombineInventory(tritium_excess);
+
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+  }
```

Comment:
Maybe put this in a function called `StoreExcessTritium()` for readability.

Reply:
Similarly, the next block could be `StoreBlanketTritium()` or something like that??